pkg/query/endpointset.go

```diff
@@ -241,29 +241,28 @@ func (c *endpointSetNodeCollector) Collect(ch chan<- prometheus.Metric) {
 // EndpointSet maintains a set of active Thanos endpoints. It is backed up by Endpoint Specifications that are dynamically fetched on
 // every Update() call.
 type EndpointSet struct {
+	now    nowFunc
 	logger log.Logger
 
 	// Endpoint specifications can change dynamically. If some component is missing from the list, we assume it is no longer
 	// accessible and we close gRPC client for it, unless it is strict.
-	endpointSpec        func() []*GRPCEndpointSpec
-	dialOpts            []grpc.DialOption
-	gRPCInfoCallTimeout time.Duration
+	endpointSpec             func() []*GRPCEndpointSpec
+	dialOpts                 []grpc.DialOption
+	gRPCInfoCallTimeout      time.Duration
+	unhealthyEndpointTimeout time.Duration
 
-	updateMtx            sync.Mutex
-	endpointsMtx         sync.RWMutex
-	endpointsStatusesMtx sync.RWMutex
+	updateMtx sync.Mutex
 
-	// Main map of stores currently used for fanout.
+	endpointsMtx    sync.RWMutex
 	endpoints       map[string]*endpointRef
 	endpointsMetric *endpointSetNodeCollector
-
-	// Map of statuses used only by UI.
-	endpointStatuses         map[string]*EndpointStatus
-	unhealthyEndpointTimeout time.Duration
 }
 
+type nowFunc func() time.Time
```

Comment:
Maybe a comment as to why this is a function instead of just `time.Time` would be great? Afaiu, this is so that we can get the exact current time while pruning endpoints and setting endpoint status. 🙂

Reply:
Yes good idea, added 👍